packages/runtime/src/next/client.ts

```diff
@@ -368,15 +385,20 @@ export class Makeswift {
 
     const response = await this.fetch(`v4/pages?${queryParams.toString()}`, siteVersion)
     if (!response.ok) {
-      console.error('Failed to get pages', await response.json())
-      throw new Error(`Failed to get pages with error: "${response.statusText}"`)
+      console.error('Failed to get pages', {
+        response: await failedResponseBody(response),
+        siteVersion,
+        params,
+      })
+
+      throw new Error(`Failed to get pages: ${responseError(response)}`)
```

Comment:
A general theme here and below is to:

1. Include all function arguments in the `console.error` message
2. Include the response status and status message in the exception, together with the resource ID (if any).